File: Semestru 4/MPP/lab/java/lab3/Services/src/main/java/org/example/services/IService.java
package org.example.services;

import org.example.domain.Child;
import org.example.domain.Event;
import org.example.domain.Signup;
import org.example.domain.LoginInfo;

import java.io.Serializable;

public interface IService extends Serializable {
    Iterable<Child> GetAllChildren();
    Child GetChildById(int id);
    Child AddChild (String name, String cnp);
    Child UpdateChild (Child child);
    Iterable<Child> GetChildByAge (int age);
    Iterable<Event> GetAllEvents();
    Event GetEventById(int id);
    Event AddEvent(String name, int minAge, int maxAge);
    Event UpdateEvent(Event event);
    LoginInfo GetByUsername (String username);
    Iterable<Signup> GetAllSignups();
    Signup AddSignup (Child child, Event event);
    Signup GetSignupById (int childId, int eventId);
    Iterable<Signup> GetSignupByEvent (int eventId);
    Iterable<Signup> GetSignupByChild (int childId);
    Iterable<Signup> GetAllSignupsMapped (IServiceChild serviceChild, IServiceEvent serviceEvent);
    Signup DeleteSignup (int childId, int eventId);
    IServiceChild getServiceChild();
    IServiceEvent getServiceEvent();
    IServiceSignup getServiceSignup();
    IServiceLogin getServiceLogin();
    boolean login(LoginInfo login, IObserver client);
    boolean logout(LoginInfo login, IObserver client);
}

File: Semestru 4/MPP/lab/.net/lab3/lab3/Services/services_async/IService.cs
using Model.app.domain;
using System;

namespace Services.services
{
	public interface IService
	{
		IEnumerable<Child> GetAllChildren();
		Child? GetChildById(int id);
		Child? AddChild(String name, String cnp);
		Child? UpdateChild(Child child);
		IEnumerable<Child>? GetChildByAge(int age);
		IEnumerable<Event> GetAllEvents();
		Event? GetEventById(int id);
		Event? AddEvent(String name, int minAge, int maxAge);
		Event? UpdateEvent(Event event_);
		IEnumerable<Event>? GetEventByMinAge(int minAge);
		IEnumerable<Event>? GetEventByMaxAge(int maxAge);
		LoginInfo GetLoginByUsername (String username);
		IEnumerable<Signup> GetAllSignups();
		Signup? AddSignup(Child child, Event event_);
		Signup? GetSignupById (int childId, int eventId);
		IEnumerable<Signup>? GetSignupByEvent(int eventId);
		IEnumerable<Signup>? GetSignupByChild(int childId);
		bool Login(LoginInfo login, IObserver client);
		bool Logout(LoginInfo login, IObserver client);
	}
}
